### ISSUE TITLE
[1.7.2] "lnfwdreport" from LND Lightning Options menu complains about "bad array subscript"

### ISSUE THREAD

**Issue by cricktor** [raspiblitz/raspiblitz#2998]
Yesterday I updated from v1.7.1 to v1.7.2 which as far as I recall went through smoothly.
My node's Settings before update: +run behind TOR; +LND; +Accept Keysend
My node's Services before update: +BTC-RPC-Explorer +Mempool Space +JoinMarket +LND RTL +LND LIT +BOS
Of course node's blockchain and LND are synced before testing.

Calling the option "lnfwdreport" from LND Lightning Options menu I get the following output:
```
/home/admin/config.scripts/lnd.fwdreport.sh: line 101: pubKeyAliasLookup: bad array subscript
/home/admin/config.scripts/lnd.fwdreport.sh: line 101: pubKeyAliasLookup: bad array subscript
/home/admin/config.scripts/lnd.fwdreport.sh: line 101: pubKeyAliasLookup: bad array subscript
/home/admin/config.scripts/lnd.fwdreport.sh: line 101: pubKeyAliasLookup: bad array subscript
/home/admin/config.scripts/lnd.fwdreport.sh: line 101: pubKeyAliasLookup: bad array subscript
/home/admin/config.scripts/lnd.fwdreport.sh: line 101: pubKeyAliasLookup: bad array subscript
/home/admin/config.scripts/lnd.fwdreport.sh: line 101: pubKeyAliasLookup: bad array subscript
/home/admin/config.scripts/lnd.fwdreport.sh: line 101: pubKeyAliasLookup: bad array subscript
Date              Channel In  Channel Out                 Amount  Fee
----------------  ----------  -----------                 ------  ---
...
```
Otherwise the data seems to be fine in the output. I am not sure what causes this, but could the reason be that I had to close one channel which was involved in received routing fees. I reopened that channel later with increased capacity prior to the update to v1.7.2. Maybe there is a more clever way to update the capacity of an open channel. I have seen some command for that in the BOS tool but I'm still new to it.

Do you need more details from me?

**Comment by rootzoll** [comment]
I take this for a quick check for the v1.8 release. Basically the values of `$channelInPubKey` and `$channelOutPubKey` should be checked not being zero should be checked before calling in 101 `${pubKeyAliasLookup[$channelInPubKey]},${pubKeyAliasLookup[$channelOutPubKey]}`

**Comment by rootzoll** [comment]
OK should be fixed for v1.8 release. Closing issue.
